src/main/java/it/reply/orchestrator/service/deployment/providers/ImServiceImpl.java

```diff
@@ -559,6 +559,15 @@ public boolean isUndeployed(DeploymentMessage deploymentMessage) {
     return false;
   }
 
+  @Override
+  public void doProviderTimeout(DeploymentMessage deploymentMessage) {
+    if (!deploymentMessage.isPollComplete()) {
```

Comment:
If we get here method must throw, so no `if` is needed

Reply:
You must try. In the previous configuration, when the timer was not on the boundary, control was necessary because the trigger, once fired, remained in that state even repeating the cycle with a new provider. The component with the 'X' had been installed and behaved like an OR rather than an AND for which the timer trigger operated the exception in any case. I can try to take it off and try it out.

Reply:
done